clouds.yaml.template

```diff
@@ -1,9 +1,24 @@
 clouds:
+{{- if eq .AuthType "none"}}
   metal3-bootstrap:
-    auth_type: none
+    auth_type: {{.AuthType}}
     baremetal_endpoint_override: {{.BootstrapIronicURL}}
     baremetal_introspection_endpoint_override: {{.BootstrapInspectorURL}}
   metal3:
-    auth_type: none
+    auth_type: {{.AuthType}}
     baremetal_endpoint_override: {{.ClusterIronicURL}}
     baremetal_introspection_endpoint_override: {{.ClusterInspectorURL}}
+{{- else if eq .AuthType "http_basic"}}
+  metal3:
+    auth_type: {{.AuthType}}
+    auth:
```

Comment:
It seems like the only part that's different between the 2 cases is the fact that for http_basic this auth section is added, so the template could be simplified to avoid repeating the rest of the settings.

Reply:
not only the `auth` section, we "split" the `metal3` cloud into `metal3` and `metal3-inspector`, in the noauth scenario we have the baremetal_introspection_endpoint_override set for `metal3` .
I will try to come up with a more simplified version

Reply:
```
clouds:
  metal3-bootstrap:
    auth_type: none
    baremetal_endpoint_override: {{.BootstrapIronicURL}}
    baremetal_introspection_endpoint_override: {{.BootstrapInspectorURL}}
  metal3:
    auth_type: {{.AuthType}}
{{- if eq .AuthType "http_basic"}}
    auth:
      username: {{.IronicUser}}
      password: {{.IronicPassword}}
    baremetal_endpoint_override: {{.ClusterIronicURL}}
  metal3-inspector:
    auth_type: {{.AuthType}}
    auth:
      username: {{.InspectorUser}}
      password: {{.InspectorPassword}}
    baremetal_introspection_endpoint_override: {{.ClusterInspectorURL}}
{{- end}}
{{- if eq .AuthType "none"}}
    baremetal_endpoint_override: {{.ClusterIronicURL}}
    baremetal_introspection_endpoint_override: {{.ClusterInspectorURL}}
{{- end}}
```

Something like this @dhellmann ?

Reply:
How about 

```
clouds:
  metal3-bootstrap:
    auth_type: none
    baremetal_endpoint_override: {{.BootstrapIronicURL}}
    baremetal_introspection_endpoint_override: {{.BootstrapInspectorURL}}
  metal3:
    baremetal_endpoint_override: {{.ClusterIronicURL}}
    auth_type: {{.AuthType}}
{{- if eq .AuthType "http_basic"}}
    auth:
      username: {{.IronicUser}}
      password: {{.IronicPassword}}
{{- end}}
  metal3-inspector:
    baremetal_introspection_endpoint_override: {{.ClusterInspectorURL}}
    auth_type: {{.AuthType}}
{{- if eq .AuthType "http_basic"}}
    auth:
      username: {{.IronicUser}}
      password: {{.IronicPassword}}
{{- end}}
```

Do we need credentials for the `metal3-bootstrap` entry?

Reply:
Ah, I couldn't tell from the template that your goal was to have 2 different implementations for the different auth types. It might be clearer to make each if statement contain a complete set of the values so the reader doesn't have to parse the template to understand it, like this:

```
clouds:
  metal3-bootstrap:
    auth_type: none
    baremetal_endpoint_override: {{.BootstrapIronicURL}}
    baremetal_introspection_endpoint_override: {{.BootstrapInspectorURL}}
{{- if eq .AuthType "none"}}
  metal3:
    auth_type: {{.AuthType}}
    baremetal_endpoint_override: {{.ClusterIronicURL}}
    baremetal_introspection_endpoint_override: {{.ClusterInspectorURL}}
{{- end}}
{{- if eq .AuthType "http_basic"}}
  metal3:
    auth_type: {{.AuthType}}
    auth:
      username: {{.IronicUser}}
      password: {{.IronicPassword}}
    baremetal_endpoint_override: {{.ClusterIronicURL}}
  metal3-inspector:
    auth_type: {{.AuthType}}
    auth:
      username: {{.InspectorUser}}
      password: {{.InspectorPassword}}
    baremetal_introspection_endpoint_override: {{.ClusterInspectorURL}}
{{- end}}
```

And if there is a comment syntax for the template, we could use that to highlight what's going on.

Reply:
ack!

Reply:
> for `metal3-bootstrap` afaik we don't need.

Not yet, but it'll be the next thing.